# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/GlobalObject.cpp
# Commit: 2186bd51bebd
# Full Hash: 2186bd51bebd0a087d2b37b58ddb590110eda236
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 4 - Move ObjectFlags from BaseShape to Shape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106974
# ==============================================================================

diff -r d5d3cb487914 -r 2186bd51bebd js/src/vm/GlobalObject.cpp
--- a/js/src/vm/GlobalObject.cpp	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/src/vm/GlobalObject.cpp	Sat Mar 06 01:05:20 2021 +0000
@@ -1082,7 +1082,8 @@
   Rooted<BaseShape*> base(cx, last->base());
 
   RootedId id(cx, NameToId(name));
-  Rooted<StackShape> child(cx, StackShape(base, id, slot, 0));
+  Rooted<StackShape> child(cx,
+                           StackShape(base, last->objectFlags(), id, slot, 0));
   Shape* shape = cx->zone()->propertyTree().getChild(cx, last, child);
   if (!shape) {
     return false;